### PR TITLE
ci: Add validate workflow instead of travis

### PR DESCRIPTION
**Pull request by MatanBobi**
**What**:

Add validate workflow by using `yo kcd-oss`

**Why**:

Because travis no more

**How**:

using `yo kcd-oss`

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Updated Type Definitions
- [ ] Ready to be merged

I guess we can reduce the number of node versions in the metrics, but I kept it as is for now..

### CODE REVIEW

**Comment by MatanBobi** [comment]
Looks like it failed on snapshots that actually passed on my local env.. any ideas?

**Comment by nickmccurdy** [comment]
Is this perhaps disabling coloring because it's in a test environment (but the tests assert that the output is colored)?

File: .github/workflows/validate.yml
Comment (nickmccurdy) [inline .github/workflows/validate.yml:6]:
```suggestion
      - 'master'
```

File: .github/workflows/validate.yml
Comment (nickmccurdy) [inline .github/workflows/validate.yml:72]:
```suggestion
              'master',
```

File: .github/workflows/validate.yml
Comment (nickmccurdy) [inline .github/workflows/validate.yml:46]:
```suggestion
      contains('refs/heads/master,refs/heads/beta,refs/heads/next,refs/heads/alpha',
```

File: other/CODE_OF_CONDUCT.md
Comment (nickmccurdy) [inline other/CODE_OF_CONDUCT.md:63]: @kentcdodds Do you want Testing Library moderation issues to be handled via your email or should we have a contact method for all community moderators (such as a shared email account, a form on Netlify, or a channel on Discord)?

**Review by nickmccurdy — CHANGES_REQUESTED** [review]

**Comment by nickmccurdy** [comment]
> Looks like it failed on snapshots that actually passed on my local env.. any ideas?

Fixed, we were missing the `FORCE_COLOR` environment variable from the Travis config.

**Comment by MatanBobi** [comment]
> > Looks like it failed on snapshots that actually passed on my local env.. any ideas?
> 
> Fixed, we were missing the `FORCE_COLOR` environment variable from the Travis config.

Thanks @nickmccurdy! missed that..

**Comment by nickmccurdy** [comment]
We can merge this if you want. If Kent doesn't want Testing Library emails sent to him directly, he can forward anything important to us and we can discuss updating our Code of Conduct on Discord.

File: other/CODE_OF_CONDUCT.md
Comment (kentcdodds) [reply other/CODE_OF_CONDUCT.md:63]: We could use TestingLibraryOSS@gmail.com which everyone has access to. I've got that address forwarding to me and others can do the same. I know a lot of people wouldn't feel comfortable chatting a COC violation on a public place like discord.

Comment (nickmccurdy) [reply other/CODE_OF_CONDUCT.md:63]: Sure, you can forward that to me. Can we still reply all with this forwarding though?

Comment (kentcdodds) [reply other/CODE_OF_CONDUCT.md]: No, I imagine that if we got a report we could talk it over together in discord (private maintainer channel) or on GitHub and then one of us could continue the conversation when we decide what to do. I don't think anything more sophisticated is necessary because it's not something we'll need to do often.

**Review by nickmccurdy — APPROVED** [review]

**Pull request merged**
